src/engine/bank02.asm

```diff
@@ -1,17 +1,1775 @@
+; works out which cursor coordinate to go to
+; and sets carry flag if A or B are pressed
+; input
+; returns a =  $1 if A pressed
+; returns a = $ff if B pressed
+HandleDuelMenuInput_PlayArea: ; 86ac (2:46ac)
+	xor a
+	ld [wcfe3], a
+	ld a, [wCursorDuelXPosition]
+	ld d, a
+	ld a, [wCursorDuelYPosition]
+	ld e, a
+
+; d = cursor x position
+; e = cursor y position
+
+	ldh a, [hDPadHeld]
+	or a
+	jr z, .skip
+
+; pad is pressed
+	ld a, [wce5e]
+	and %10000000
+	ldh a, [hDPadHeld]
+	jr nz, .check_vertical
+	bit D_LEFT_F, a ; test left button
+	jr nz, .horizontal
+	bit D_RIGHT_F, a ; test right button
+	jr z, .check_vertical
+
+; handle horizontal input
+.horizontal
+	ld a, [wce5e]
+	and %01111111
+	or a
+	jr nz, .asm_86dd ; jump if wce5e's lower 7 bits aren't set
+	ld a, e
+	or a
+	jr z, .flip_x ; jump if y is 0
+
+; wce5e = %10000000
+; e = 1
+	dec e ; change y position
+	jr .flip_x
+
+.asm_86dd
+	ld a, e
+	or a
+	jr nz, .flip_x ; jump if y is not 0
+	inc e ; change y position
+.flip_x
+	ld a, d
+	xor $01 ; flip x position
+	ld d, a
+	jr .erase
+
+.check_vertical
+	bit D_UP_F, a
+	jr nz, .vertical
+	bit D_DOWN_F, a
+	jr z, .skip
+
+; handle vertical input
+.vertical
+	ld a, d
+	or a
+	jr z, .flip_y ; jump if x is 0
+	dec d
+.flip_y
+	ld a, e
+	xor $01 ; flip y position
+	ld e, a
+
+.erase
+	ld a, $01
+	ld [wcfe3], a
+	push de
+	call DrawCursorEmpty_OppPlayArea
+	pop de
+
+;update x and y cursor positions
+	ld a, d
+	ld [wCursorDuelXPosition], a
+	ld a, e
+	ld [wCursorDuelYPosition], a
+
+; reset cursor blink
+	xor a
+	ld [wDuelCursorBlinkCounter], a
+
+.skip
+	ldh a, [hKeysPressed]
+	and A_BUTTON | B_BUTTON
+	jr z, .sfx
+	and A_BUTTON
+	jr nz, .a_pressed
+
+; B pressed
+	ld a, $ff ; cancel
+	call PlaySFXConfirmOrCancel
+	scf
+	ret
+	
+.a_pressed
+	call DrawCursor_OppPlayArea
+	ld a, $01
+	call PlaySFXConfirmOrCancel
+	scf
+	ret
+
+.sfx
+	ld a, [wcfe3]
+	or a
+	jr z, .draw_cursor
+	call PlaySFX
+
+.draw_cursor
+	ld hl, wDuelCursorBlinkCounter
+	ld a, [hl]
+	inc [hl]
+	and %00001111
+	ret nz ; only update cursor if blink's lower nibble is 0
+
+	ld a, $0f ; cursor byte
+	bit 4, [hl] ; only draw cursor if blink counter's fourth bit is not set
+	jr z, DrawByteInCursor_OppPlayArea
+; fallthrough
+
+; transforms cursor position into coordinates
+; in order to draw byte on menu cursor
+DrawCursorEmpty_OppPlayArea: ; 8741 (2:4741)
+	ld a, $00 ; white tile
+; fallthrough
+
+; draws in the cursor position
+; input:
+; a = tile byte to draw
+DrawByteInCursor_OppPlayArea: ; 8743 (2:4743)
+	ld e, a
+	ld a, 10
+	ld l, a
+	ld a, [wCursorDuelXPosition]
+	ld h, a
+	call HtimesL
+; h = 10 * cursor x pos
+
+	ld a, l
+	add 1
+	ld b, a
+	ld a, [wCursorDuelYPosition]
+	sla a
+	add 14
+	ld c, a
+; c = 11 + 2 * cursor y pos + 14
 
-Func_8211: ; 8211 (2:4211)
-	INCROM $8211, $833c
+; draw tile loaded in e
+	ld a, e
+	call WriteByteToBGMap0
+	or a
+	ret
 
-Func_833c: ; 833c (2:433c)
-	INCROM $833c, $8764
+DrawCursor_OppPlayArea: ; 8760 (2:4760)
```

Comment:
Same as 30 lines above. This is also counterpart to ``Func_90f7`` in the same way as ``DrawCheckMenuCursor_YourOrOppPlayArea`` is to ``DrawCheckMenuCursor`` and ``EraseCheckMenuCursor_YourOrOppPlayArea`` is to ``EraseCheckMenuCursor``. For this one, since it activates the cursor, you could just use ``DisplayCheckMenuCursor_YourOrOppPlayArea``, or whatever you want.